### PR TITLE
Add placement group function to terraform

### LINKED ISSUE #3456: Add Placement Group support for PowerVS
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In June 2021 the PowerVS REST API added support for Placement Groups:

* https://cloud.ibm.com/apidocs/power-cloud#release-notes

Documented here:

* https://cloud.ibm.com/apidocs/power-cloud#pcloud-placementgroups-post

These would be a useful addition to the Terraform provider.

### New or Affected Resource(s)

* ibm_pi_placement_group
* ibm_pi_instance

### Potential Terraform Configuration

Proposed new resource:

```hcl
resource "ibm_pi_placement_group" "placement_group" {
    pi_cloud_instance_id    = "51e1879c-bcbe-4ee1-a008-49cdba0eaf60"
    pi_placement_name       = "some_group_name"
    pi_placement_policy     = "affinity"
}
```

Proposed new data source:

```hcl
data "ibm_pi_placement_group" "ds_placement_group" {
    pi_cloud_instance_id    = "51e1879c-bcbe-4ee1-a008-49cdba0eaf60"
    pi_placement_group_name = "some_group_name"
}
```

Addition to existing ibm_pi_instance resource:

```hcl
resource "ibm_pi_instance" "test-instance" {
    pi_memory               = "4"
    pi_processors           = "2"
    pi_instance_name        = "test-vm"
    pi_proc_type            = "shared"
    pi_image_id             = "${data.ibm_pi_image.powerimages.id}"
    pi_key_pair_name        = ibm_pi_key.key.key_id
    pi_sys_type             = "s922"
    pi_cloud_instance_id    = "51e1879c-bcbe-4ee1-a008-49cdba0eaf60"
    pi_pin_policy           = "none"
    pi_health_status        = "WARNING"
    pi_network {
      network_id = data.ibm_pi_public_network.dsnetwork.id
    }

# NEW VARIABLE

    pi_placement_group_id   = "49fba6c9-23f8-40bc-9899-aca322ee7d5b"
}
```

### References

* https://cloud.ibm.com/apidocs/power-cloud#release-notes
* https://cloud.ibm.com/apidocs/power-cloud#pcloud-placementgroups-post

### PR DESCRIPTION
**Pull request by rmoralesjr**
Add placement group function to terraform which
includes a new placement group field the instance
object and a new placement group resource.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #3456

Output from acceptance testing:

=== RUN   TestAccIBMPIPlacementGroupBasic
[INFO] TEST PG MEMBERS: []string{}
[INFO] TEST PG name: "tf-pi-placement-group-36"
[INFO] TEST PG policy: "affinity"
[INFO] TEST PG ID: "74af946b-1b84-4d1b-8dd8-c3334b600f04"
[INFO] TEST If PG is updated wth instance member: a5e99147-4520-4380-bbe0-db6cf8b67f21
[INFO] TEST Instance ID: a5e99147-4520-4380-bbe0-db6cf8b67f21
[INFO] TEST If PG removed instance member. Instance members count=: 0
[INFO] TEST Instance ID that should be removed: a5e99147-4520-4380-bbe0-db6cf8b67f21
[INFO] TEST If PG members field is updated wth instance member: ed77ff3b-f9e6-42d1-b157-8a6913e089a4
[INFO] TEST If PG members field is updated wth instance member: a5e99147-4520-4380-bbe0-db6cf8b67f21
[INFO] TEST Instance ID: a5e99147-4520-4380-bbe0-db6cf8b67f21
[INFO] TEST new Instance ID: ed77ff3b-f9e6-42d1-b157-8a6913e089a4
[INFO] TEST Instance ID: a5e99147-4520-4380-bbe0-db6cf8b67f21
[INFO] TEST new Instance ID: ed77ff3b-f9e6-42d1-b157-8a6913e089a4
[INFO] TEST If PG was removed from instance tf-pi-placement-group-36 who was a member
[INFO] TEST If PG was removed from instance tf-pi-placement-group-36-in-pg who was a member
--- PASS: TestAccIBMPIPlacementGroupBasic (1386.86s)
PASS

=== RUN   TestAccIBMPIPlacementGroupDataSource_basic
--- PASS: TestAccIBMPIPlacementGroupDataSource_basic (8.18s)
PASS

### CODE REVIEW

File: ibm/data_source_ibm_pi_placement_group.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+			helpers.PICloudInstanceId: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			PIPlacementGroupMembers: {
+				Type:     schema.TypeList,
+				Optional: true,
```
Comment (yussufsh) [inline ibm/data_source_ibm_pi_placement_group.go:46]: This is not an input argument, you can remove the Optional line.

File: ibm/data_source_ibm_pi_placement_group.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceIBMPIPlacementGroup() *schema.Resource {
+
+	return &schema.Resource{
+		ReadContext: dataSourceIBMPIPlacementGroupRead,
+		Schema: map[string]*schema.Schema{
+			helpers.PIPlacementGroupID: {
```
Comment (yussufsh) [inline ibm/data_source_ibm_pi_placement_group.go:23]: Isn't `PIPlacementGroupName` as required and `PIPlacementGroupID` as computed?

File: ibm/data_source_ibm_pi_placement_group.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	sess, err := meta.(ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	powerinstanceid := d.Get(helpers.PICloudInstanceId).(string)
+	placementGroupid := d.Get(helpers.PIPlacementGroupID).(string)
+	client := st.NewIBMPIPlacementGroupClient(ctx, sess, powerinstanceid)
+
+	response, err := client.Get(placementGroupid)
```
Comment (yussufsh) [inline ibm/data_source_ibm_pi_placement_group.go:65]: You should be passing the name instead of the id.

File: ibm/resource_ibm_pi_instance.go [hunk 4]
```diff
@@ -656,6 +664,39 @@ func resourceIBMPIInstanceUpdate(ctx context.Context, d *schema.ResourceData, me
...
+
+		if len(strings.TrimSpace(old)) > 0 && old != "none" {
+			placementGroupID := old
+			//remove server from old placement group
+			body := &models.PlacementGroupServer{
+				ID: &instanceID,
+			}
+			_, err := pgClient.DeleteMember(placementGroupID, body)
+			if err != nil {
+				return diag.Errorf("failed to perform the placement group delete operation... %v", err)
```
Comment (yussufsh) [inline ibm/resource_ibm_pi_instance.go:683]: Just return the err, `DeleteMember` has the proper message with wrapped err.

File: ibm/resource_ibm_pi_instance.go [hunk 5]
```diff
@@ -656,6 +664,39 @@ func resourceIBMPIInstanceUpdate(ctx context.Context, d *schema.ResourceData, me
...
+
+		if len(strings.TrimSpace(new)) > 0 && new != "none" {
+			placementGroupID := new
+			// add server to a new placement group
+			body := &models.PlacementGroupServer{
+				ID: &instanceID,
+			}
+			_, err := pgClient.AddMember(placementGroupID, body)
+			if err != nil {
+				return diag.Errorf("failed to add placement group member... %v", err)
```
Comment (yussufsh) [inline ibm/resource_ibm_pi_instance.go:695]: Just return the err, `AddMember` has the proper message with wrapped err.

File: ibm/resource_ibm_pi_instance.go [hunk 6]
```diff
@@ -1167,6 +1208,10 @@ func createPVMInstance(d *schema.ResourceData, client *st.IBMPIInstanceClient, i
 		body.StorageConnection = sc.(string)
 	}
 
+	if d.Get(PIInstancePlacementGroupID) != nil {
```
Comment (yussufsh) [inline ibm/resource_ibm_pi_instance.go:1211]: Can you use `d.GetOk` instead?

File: ibm/resource_ibm_pi_placement_group.go [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+			helpers.PICloudInstanceId: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "PI cloud instance ID",
+			},
+
+			PIPlacementGroupMembers: {
+				Type:        schema.TypeSet,
+				Optional:    true,
```
Comment (yussufsh) [inline ibm/resource_ibm_pi_placement_group.go:61]: Remove this line.

File: ibm/resource_ibm_pi_placement_group.go [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+	client := st.NewIBMPIPlacementGroupClient(ctx, sess, powerinstanceid)
+	body := &models.PlacementGroupCreate{
+		Name:   &name,
+		Policy: &policy,
+	}
+
+	response, err := client.Create(body)
+	if err != nil {
+		log.Printf("[DEBUG]  err %s", err)
+		return diag.Errorf("Failed to create the placement group %v", err)
```
Comment (yussufsh) [inline ibm/resource_ibm_pi_placement_group.go:94]: Return the err.

File: website/docs/d/pi_placement_group.html.markdown [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+## Argument reference
+Review the argument references that you can specify for your data source. 
+
+- `pi_cloud_instance_id` - (Required, String) The GUID of the service instance associated with an account.
+- `pi_placement_group_name` - (Required, String) The name of the placement group.
+
+## Attribute reference
+In addition to all argument reference list, you can access the following attribute references after your data source is created. 
+
+- `pi_placement_group_policy` - (String) The value of the group's affinity policy. Valid values are affinity and anti-affinity.
```
Comment (yussufsh) [inline website/docs/d/pi_placement_group.html.markdown:47]: Also add the `id`.

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: ibm/data_source_ibm_pi_placement_group.go (hunk 2 above)
Comment (rmoralesjr) [reply ibm/data_source_ibm_pi_placement_group.go:23]: I made this suggested change.

File: ibm/data_source_ibm_pi_placement_group.go (hunk 1 above)
Comment (rmoralesjr) [reply ibm/data_source_ibm_pi_placement_group.go:46]: I made this suggested change.

File: ibm/data_source_ibm_pi_placement_group.go (hunk 3 above)
Comment (rmoralesjr) [reply ibm/data_source_ibm_pi_placement_group.go:65]: I made this suggested change.

File: ibm/resource_ibm_pi_instance.go (hunk 4 above)
Comment (rmoralesjr) [reply ibm/resource_ibm_pi_instance.go:683]: I made this suggested change.

File: ibm/resource_ibm_pi_instance.go (hunk 5 above)
Comment (rmoralesjr) [reply ibm/resource_ibm_pi_instance.go:695]: I made this suggested change.

File: ibm/resource_ibm_pi_instance.go (hunk 6 above)
Comment (rmoralesjr) [reply ibm/resource_ibm_pi_instance.go:1211]: I made this suggested change.

File: ibm/resource_ibm_pi_placement_group.go (hunk 7 above)
Comment (rmoralesjr) [reply ibm/resource_ibm_pi_placement_group.go:61]: I made this suggested change.

File: ibm/resource_ibm_pi_placement_group.go (hunk 8 above)
Comment (rmoralesjr) [reply ibm/resource_ibm_pi_placement_group.go:94]: I made this suggested change.

File: website/docs/d/pi_placement_group.html.markdown (hunk 9 above)
Comment (rmoralesjr) [reply website/docs/d/pi_placement_group.html.markdown:47]: I made this suggested change.

**Comment by rmoralesjr** [comment]
Here are the test run results after the code review changes
  === RUN   TestAccIBMPIPlacementGroupBasic
[INFO] TEST PG MEMBERS: []string{}
[INFO] TEST PG name: "tf-pi-placement-group-83"
[INFO] TEST PG policy: "affinity"
[INFO] TEST PG ID: "338724ab-58da-43bf-820d-de75798401ee"
[INFO] TEST If PG is updated wth instance member: 8eba54ee-6cc4-4893-8bb2-22658ddc0823
[INFO] TEST Instance ID: 8eba54ee-6cc4-4893-8bb2-22658ddc0823
[INFO] TEST If PG removed instance member. Instance members count=: 0
[INFO] TEST Instance ID that should be removed: 8eba54ee-6cc4-4893-8bb2-22658ddc0823
[INFO] TEST If PG members field is updated wth instance member: 48f9ccd7-5692-4996-9319-8f53837575fa
[INFO] TEST If PG members field is updated wth instance member: 8eba54ee-6cc4-4893-8bb2-22658ddc0823
[INFO] TEST Instance ID: 8eba54ee-6cc4-4893-8bb2-22658ddc0823
[INFO] TEST new Instance ID: 48f9ccd7-5692-4996-9319-8f53837575fa
[INFO] TEST Instance ID: 8eba54ee-6cc4-4893-8bb2-22658ddc0823
[INFO] TEST new Instance ID: 48f9ccd7-5692-4996-9319-8f53837575fa
[INFO] TEST If PG was removed from instance tf-pi-placement-group-83 who was a member
[INFO] TEST If PG was removed from instance tf-pi-placement-group-83-in-pg who was a member
--- PASS: TestAccIBMPIPlacementGroupBasic (1254.75s)
PASS

=== RUN   TestAccIBMPIPlacementGroupDataSource_basic
--- PASS: TestAccIBMPIPlacementGroupDataSource_basic (10.40s)
PASS

File: ibm/data_source_ibm_pi_placement_group.go [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+	}
+
+	cloudInstanceID := d.Get(helpers.PICloudInstanceId).(string)
+	placementGroupName := d.Get(helpers.PIPlacementGroupName).(string)
+	client := st.NewIBMPIPlacementGroupClient(ctx, sess, cloudInstanceID)
+
+	response, err := client.Get(placementGroupName)
+	if err != nil {
+		log.Printf("[DEBUG]  err %s", err)
+		return diag.Errorf("Failed to get the placement group %v", err)
```
Comment (yussufsh) [inline ibm/data_source_ibm_pi_placement_group.go:67]: No need for adding the message here as it is already part of err created by the client.

File: ibm/resource_ibm_pi_placement_group.go [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+
+func resourceIBMPIPlacementGroupRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	sess, err := meta.(ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	parts, err := idParts(d.Id())
+	if err != nil {
+		return diag.Errorf("Failed to get the placement group %v", err)
```
Comment (yussufsh) [inline ibm/resource_ibm_pi_placement_group.go:110]: This is not GET failure. You can directly return the error.

File: ibm/resource_ibm_pi_placement_group.go [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+	}
+
+	cloudInstanceID := parts[0]
+	client := st.NewIBMPIPlacementGroupClient(ctx, sess, cloudInstanceID)
+
+	response, err := client.Get(parts[1])
+	if err != nil {
+		log.Printf("[DEBUG]  err %s", err)
+		return diag.Errorf("Failed to get the placement group %v", err)
```
Comment (yussufsh) [inline ibm/resource_ibm_pi_placement_group.go:119]: Return directly.

File: website/docs/d/pi_instance.html.markdown [hunk 13]
```diff
@@ -70,3 +70,4 @@ In addition to all argument reference list, you can access the following attribu
 - `storage_type` - (String) Storage type where server is deployed.
 - `virtual_cores_assigned` - (Integer) The virtual cores that are assigned to the instance.
 - `volumes`- (List of strings) The list of volume IDs that are attached to the instance.
+- `placement_group_id`- (String) The ID of the placement group that the instance is in if any.
```
Comment (yussufsh) [inline website/docs/d/pi_instance.html.markdown:73]: Need to sort the attribute names alphabetically in all the doc changes.

File: website/docs/r/pi_placement_group.html.markdown [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+
+- `pi_cloud_instance_id` - (Required, String) The GUID of the service instance associated with an account.
+- `pi_placement_group_name`  - (Required, String) The name of the placement group. 
+- `pi_placement_group_policy` - (Required, String) The value of the group's affinity policy. Valid values are affinity and anti-affinity. 
+
+
+## Attribute reference
+ In addition to all argument reference list, you can access the following attribute reference after your resource is created.
+
+- `id` - (String) The unique identifier of the key. The ID is composed of `<power_instance_id>/<key_id>`.
```
Comment (yussufsh) [inline website/docs/r/pi_placement_group.html.markdown:58]: There is no `key_id` in this resource.
Add description for `helpers.PIPlacementGroupID` and `PIPlacementGroupMembers`.

**Review by yussufsh — CHANGES_REQUESTED** [review]
Minor changes needed.

File: ibm/data_source_ibm_pi_placement_group.go (hunk 10 above)
Comment (rmoralesjr) [reply ibm/data_source_ibm_pi_placement_group.go:67]: Done.

File: ibm/resource_ibm_pi_placement_group.go (hunk 11 above)
Comment (rmoralesjr) [reply ibm/resource_ibm_pi_placement_group.go:110]: Done.

File: ibm/resource_ibm_pi_placement_group.go (hunk 12 above)
Comment (rmoralesjr) [reply ibm/resource_ibm_pi_placement_group.go:119]: Done.

File: website/docs/d/pi_instance.html.markdown (hunk 13 above)
Comment (rmoralesjr) [reply website/docs/d/pi_instance.html.markdown:73]: Done.

File: website/docs/r/pi_placement_group.html.markdown (hunk 14 above)
Comment (rmoralesjr) [reply website/docs/r/pi_placement_group.html.markdown:58]: Done.

File: ibm/data_source_ibm_pi_placement_group.go [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+func dataSourceIBMPIPlacementGroup() *schema.Resource {
+
+	return &schema.Resource{
+		ReadContext: dataSourceIBMPIPlacementGroupRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
```
Comment (yussufsh) [inline ibm/data_source_ibm_pi_placement_group.go:27]: You don't want to define this block.

File: ibm/data_source_ibm_pi_instance.go [hunk 16]
```diff
@@ -180,6 +185,11 @@ func dataSourceIBMPIInstancesRead(ctx context.Context, d *schema.ResourceData, m
 	d.Set("storage_type", powervmdata.StorageType)
 	d.Set("storage_pool", powervmdata.StoragePool)
 	d.Set("license_repository_capacity", powervmdata.LicenseRepositoryCapacity)
+	if len(strings.TrimSpace(*powervmdata.PlacementGroup)) > 0 && *powervmdata.PlacementGroup == "none" {
```
Comment (yussufsh) [inline ibm/data_source_ibm_pi_instance.go:188]: You can directly set the value to `placement_group_id`. No need to set `""`.

File: ibm/data_source_ibm_pi_placement_group.go [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+	client := st.NewIBMPIPlacementGroupClient(ctx, sess, cloudInstanceID)
+
+	response, err := client.Get(placementGroupName)
+	if err != nil {
+		log.Printf("[DEBUG]  err %s", err)
+		return diag.FromErr(err)
+	}
+
+	d.SetId(*response.ID)
+	d.Set(helpers.PIPlacementGroupName, response.Name)
```
Comment (yussufsh) [inline ibm/data_source_ibm_pi_placement_group.go:71]: Can delete this line.

File: ibm/data_source_ibm_pi_instance.go (hunk 16 above)
Comment (yussufsh) [reply ibm/data_source_ibm_pi_instance.go:188]:
```
	if *powervmdata.PlacementGroup != "none" {
		d.Set("placement_group_id", powervmdata.PlacementGroup)
	}
```

File: ibm/data_source_ibm_pi_instance.go [hunk 18]
```diff
@@ -141,6 +142,10 @@ func dataSourceIBMPIInstance() *schema.Resource {
 				Type:     schema.TypeInt,
 				Computed: true,
 			},
+			"placement_group_id": {
```
Comment (yussufsh) [inline ibm/data_source_ibm_pi_instance.go:145]: Can re-use the constant here `PIPlacementGroupID`

File: ibm/data_source_ibm_pi_instance.go [hunk 19]
```diff
@@ -180,6 +185,11 @@ func dataSourceIBMPIInstancesRead(ctx context.Context, d *schema.ResourceData, m
 	d.Set("storage_type", powervmdata.StorageType)
 	d.Set("storage_pool", powervmdata.StoragePool)
 	d.Set("license_repository_capacity", powervmdata.LicenseRepositoryCapacity)
+	if len(strings.TrimSpace(*powervmdata.PlacementGroup)) > 0 && *powervmdata.PlacementGroup == "none" {
+		d.Set("placement_group_id", "")
+	} else {
+		d.Set("placement_group_id", powervmdata.PlacementGroup)
```
Comment (yussufsh) [inline ibm/data_source_ibm_pi_instance.go:191]: Use the const `PIPlacementGroupID`.

File: ibm/data_source_ibm_pi_placement_group.go [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+				Computed: true,
+			},
+
+			helpers.PICloudInstanceId: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"members": {
```
Comment (yussufsh) [inline ibm/data_source_ibm_pi_placement_group.go:44]: Re-use const `PIPlacementGroupMembers`.

File: ibm/data_source_ibm_pi_placement_group.go [hunk 21]
```diff
@@ -0,0 +1,76 @@
...
+	response, err := client.Get(placementGroupName)
+	if err != nil {
+		log.Printf("[DEBUG]  err %s", err)
+		return diag.FromErr(err)
+	}
+
+	d.SetId(*response.ID)
+	d.Set(helpers.PIPlacementGroupName, response.Name)
+	d.Set("policy", response.Policy)
+	d.Set("members", response.Members)
```
Comment (yussufsh) [inline ibm/data_source_ibm_pi_placement_group.go:73]: `PIPlacementGroupMembers`

File: website/docs/d/pi_instance.html.markdown [hunk 22]
```diff
@@ -63,6 +63,7 @@ In addition to all argument reference list, you can access the following attribu
   - `network_id` - (String) The network ID of the instance.
   - `network_name` - (String) The network name of the instance.
   - `type` - (String) The type of the network.
+- `placement_group_id`- (String) The ID of the placement group that the instance is in if any or empty quotes `""` when not in a placement group.
```
Comment (yussufsh) [inline website/docs/d/pi_instance.html.markdown:66]: No need to mention about not providing placement group.
`The placement group for the server`

**Review by yussufsh — CHANGES_REQUESTED** [review]
Also we need a data source for fetching all the placement groups within a cloud instance.

File: ibm/data_source_ibm_pi_placement_group.go (hunk 15 above)
Comment (rmoralesjr) [reply ibm/data_source_ibm_pi_placement_group.go:27]: Done

File: ibm/data_source_ibm_pi_instance.go (hunk 16 above)
Comment (rmoralesjr) [reply ibm/data_source_ibm_pi_instance.go:188]: Done

File: ibm/data_source_ibm_pi_placement_group.go (hunk 17 above)
Comment (rmoralesjr) [reply ibm/data_source_ibm_pi_placement_group.go:71]: Done

File: ibm/data_source_ibm_pi_instance.go (hunk 18 above)
Comment (rmoralesjr) [reply ibm/data_source_ibm_pi_instance.go:145]: Done

File: ibm/data_source_ibm_pi_instance.go (hunk 19 above)
Comment (rmoralesjr) [reply ibm/data_source_ibm_pi_instance.go:191]: Done

File: ibm/data_source_ibm_pi_placement_group.go (hunk 20 above)
Comment (rmoralesjr) [reply ibm/data_source_ibm_pi_placement_group.go:44]: Done

File: ibm/data_source_ibm_pi_placement_group.go (hunk 21 above)
Comment (rmoralesjr) [reply ibm/data_source_ibm_pi_placement_group.go:73]: Done

File: website/docs/d/pi_instance.html.markdown (hunk 22 above)
Comment (rmoralesjr) [reply website/docs/d/pi_instance.html.markdown:66]: Done

**Review by yussufsh — APPROVED** [review]
lgtm

File: website/docs/r/pi_placement_group.html.markdown [hunk 23]
```diff
@@ -0,0 +1,70 @@
...
+# ibm_pi_placement_group
+Create or delete a placement group.
+
+## Example usage
+The following example enables you to create a placement group with a group policy of affinity:
+
+```terraform
+resource "ibm_pi_placement_group" "testacc_placement_group" {
+	pi_placement_group_name   = "my_pg"
+	pi_placement_group_policy = "affinity"
```
Comment (hkantare) [inline website/docs/r/pi_placement_group.html.markdown:19]: Format this example section

File: ibm/resource_ibm_pi_placement_group.go [hunk 24]
```diff
@@ -0,0 +1,153 @@
...
+
+			helpers.PIPlacementGroupName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the placement group",
+			},
+
+			helpers.PIPlacementGroupPolicy: {
+				Type:        schema.TypeString,
+				Required:    true,
```
Comment (hkantare) [inline ibm/resource_ibm_pi_placement_group.go:49]: Can we add validation here to check the Valid values are `affinity` and `anti-affinity`

File: website/docs/r/pi_placement_group.html.markdown (hunk 23 above)
Comment (rmoralesjr) [reply website/docs/r/pi_placement_group.html.markdown:19]: OK I see the properties have different indents so I'll fix that. Thanks.

File: ibm/resource_ibm_pi_placement_group.go (hunk 24 above)
Comment (rmoralesjr) [reply ibm/resource_ibm_pi_placement_group.go:49]: Done.

**Review by hkantare — APPROVED** [review]

**Pull request merged**